src/Service.Tests/CosmosTests/QueryFilterTests.cs

```diff
@@ -642,6 +647,135 @@ public async Task TestFilterOnInnerNestedFields()
             await ExecuteAndValidateResult(_graphQLQueryName, gqlQuery, dbQuery);
         }
 
+        #region Field Level Auth
+        /// <summary>
+        /// Tests that the field level query filter succeeds requests when filter fileds are authorized
+        /// </summary>
+        [TestMethod]
+        public async Task TestQueryFilterFieldAuth_AuthorizedField()
+        {
+            string gqlQuery = @"{
+                earths(first: 1, " + QueryBuilder.FILTER_FIELD_NAME + @" : {id : {eq : """ + _idList[0] + @"""}})
+                { 
+                    items {
+                        id
+                    }
+                }
+            }";
+
+            string dbQuery = $"SELECT top 1 c.id FROM c where c.id = \"{_idList[0]}\"";
+            await ExecuteAndValidateResult("earths", gqlQuery, dbQuery);
+        }
+
+        /// <summary>
+        /// Tests that the field level query filter fails requests when filter fileds are unauthorized
+        /// </summary>
+        [TestMethod]
+        public async Task TestQueryFilterFieldAuth_UnauthorizedField()
+        {
+            // Run query
+            string gqlQuery = @"{
+                earths(first: 1, " + QueryBuilder.FILTER_FIELD_NAME + @" : {name : {eq : ""test name""}})
+                { 
+                    items {
+                        name
+                    }
+                }
+            }";
+            string clientRoleHeader = AuthorizationType.Anonymous.ToString();
+            JsonElement response = await ExecuteGraphQLRequestAsync(
+                queryName: "earths",
+                query: gqlQuery,
+                variables: new() { { "name", "test name" } },
+                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: clientRoleHeader),
+                clientRoleHeader: clientRoleHeader);
+
+            // Validate the result contains the GraphQL authorization error code.
+            string errorMessage = response.ToString();
+            Assert.IsTrue(errorMessage.Contains(DataApiBuilderException.GRAPHQL_FILTER_FIELD_AUTHZ_FAILURE));
+        }
+
+        /// <summary>
+        /// Tests that the field level query filter succeeds requests when filter fileds are authorized
```

Comment:
```suggestion
        /// Tests that the field level query filter passes authorization when filter fields are authorized
        /// because all fields on object type 'planets' are included fields of the read operation 
        /// permissions defined for the anonymous role.
```

Reply:
udpated